File: api/stockmanagementapi/Models/StockItems/Lookup/StockItemLookup.cs
namespace stockmanagementapi.Models.StockItems.Lookup
{
  public class StockItemLookup
  {
    public int StockItemId { get; set; }

    public string Colour { get; set; } = string.Empty;

    public string Make { get; set; } = string.Empty;

    public string Model { get; set; } = string.Empty;

    public string VINNumber { get; set; } = string.Empty;

    public string RegistrationNumber { get; set; } = string.Empty;

    public decimal Price { get; set; }

    public decimal CostPrice { get; set; }

    public int KMS { get; set; }

    public int ModelYear { get; set; }
  }
}

File: frontend/stockmanagementapp/src/app/main-feed/Models/StockItems/stockItemLookup.ts
import { ColourEnum } from "../../Enums/ColourEnum";

export interface PagedStockItems<T> {
    items: T[];
    pageNumber: number;
    pageSize: number;
    totalItems: number;
}
 
export interface StockItemLookup {
    stockItemId: number;
    registrationNumber: string;
    make: string;
    model: string;
    modelYear: number;
    kms: number;
    Colour: ColourEnum,
    vinNumber: string,
    price: number,
    costPrice: number
}

export function isValidModelYear(year: number): boolean {
    return year >= 1900 && year <= 2100;
}